pdk/1.3/modules/ROOT/pages/policies-pdk-configure-features-contracts.adoc

```diff
@@ -0,0 +1,300 @@
+        Ok(client_data) => client_data,
+        Err(e) => {
+            logger::info!("Invalid authentication: {e}");
+            return Flow::Break(unauthorized_response("Invalid authentication", 403));
+        }
+    };
+
+    // Update the current authentication
+    if let Some(mut auth) = authentication.authentication() {
+        auth.client_id = Some(client_data.client_id);
+        auth.client_name = Some(client_data.client_name);
+
+        authentication.set_authentication(Some(&auth));
+    }
+
+    Flow::Continue(())
+}
+----
+
+This code snippet provides a simple authorization request filter:
+
+[source,Rust]
+----
+async fn my_authorization_filter(
+  state: RequestHeadersState,
+  validator: &ContractValidator,
+) -> Flow<()> {
+
+    // Extract client id with a user defined helper `extract_client_id()` function,
+    let client_id = match extract_client_id(&state) {
+        Ok(credentials) => credentials,
+        Err(e) => {
+            logger::info!("Invalid credentials: {e}");
+
+            // For simplicity, we are using a user-defined `unathorized_response()`
+            // helper function for building responses.
+            return Flow::Break(unauthorized_response("Invalid credentials", 401));
+        }
+    };
+
+    // Validate authorization
+    let validation = validator.authorize(&client_id);
+
+    if let  Err(e) = validation {
+        logger::info!("Invalid authentication: {e}");
+        return Flow::Break(unauthorized_response("Invalid authentication", 403));
+    }
+
+    Flow::Continue(())
+}
+----
+
+[[custom-credentials-extraction]]
+== Implement Custom Extraction for Client credentials
+
+Invoke the `ClientId::new()` and `ClientSecret::new()` methods to extract credentials:
+
+[source,Rust]
+----
+impl ClientId {
+   pub fn new(client_id: String) -> Self;
+}
+
+impl ClientSecret {
+   pub fn new(client_secret: String) -> Self;
+}
+----
+
+After extracting the credentials, initialize a set of the credentials:
+
+[source,Rust]
+----
+
+fn initialize_credentials(raw_client_id: String, raw_client_secret) -> (ClientId, ClientSecret) {
+   let client_id = ClientId::new(raw_client_id);
+   let client_secret = ClientSecret::new(raw_client_secret);
+
+   (client_id, client_secret)
+}
+
+----
+
+NOTE: For security, the content of `ClientSecret` is deleted after use, and the `Debug` trait does not expose the `ClientSecret` content.
+
+[[contract-validator-injection]]
+== Inject the ContractValidator Object
+
+Inject the `ContractValidator` object into the `#[entrypoint]` function and share its reference to the request filter:
+
+[source,Rust]
+----
+#[entrypoint]
+async fn configure(launcher: Launcher, validator: ContractValidator) -> Result<(), LaunchError> {
+
+    let filter = on_request(|state, authentication| my_authentication_filter(state, authentication, &validator));
+    launcher.launch(filter).await?;
+
+    OK(())
+}
+----
+
+[[contract-database-polling]]
+=== Local Contracts Database Polling
```

Comment:
```suggestion
=== Polling the Local Contracts Database
```

Reply:
Or rather, what is the user task? Maintain a valid copy of the contracts? Keep contracts up-to-date?